dwst/scripts/particles.js

```diff
@@ -20,7 +20,32 @@ const specialChars = [
   '\\',
 ];
 
-export class InvalidParticles extends Error { }
+export class InvalidParticles extends Error {
+  constructor(expression, expected, expectedStringLength, remainder) {
+    super();
+    this.expression = expression;
+    this.expected = expected;
+    // this.got = got;
+    this.remainder = remainder;
+    this.expectedStringLength = expectedStringLength;
+  }
+  getGot() {
```

Comment:
You could do `get got() {` if you want this to be a property rather than a function.

Reply:
got rid of getGot